font-face src list fails early if component fails

vitorroriz opened the pull request:
#### 6782d34ec7c624ea2eb55b5a927cd86df00bcf2c
<pre>
font-face src list fails early if component fails
<a href="https://bugs.webkit.org/show_bug.cgi?id=250332">https://bugs.webkit.org/show_bug.cgi?id=250332</a>
rdar://104039050

Reviewed by Tim Nguyen.

If one component of the list fails we should not immediately fail the parsing
of font src, since one valid component will validate the descriptor.
Therefore, we should just fail the src descriptor parsing if all components
have failed.

&quot;A component with a parse error does not invalidate the entire descriptor if there&apos;s some other valid component present.&quot;
Reference: <a href="https://github.com/web-platform-tests/wpt/blob/e9470cde64/css/css-fonts/parsing/font-face-src-list.html">https://github.com/web-platform-tests/wpt/blob/e9470cde64/css/css-fonts/parsing/font-face-src-list.html</a>

* LayoutTests/fast/css/font-face-src-parsing-expected.txt:
* LayoutTests/fast/css/font-face-src-parsing.html:

* LayoutTests/fast/css/font-face-unquoted-local-expected.html: Removed.
* LayoutTests/fast/css/font-face-unquoted-local-expected.txt: Added.
* LayoutTests/fast/css/font-face-unquoted-local.html:
This test was making use of faulty behavior of src list failing early to build the test.

* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-face-src-format-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-face-src-list-expected.txt:
* Source/WebCore/css/parser/CSSPropertyParserWorkerSafe.cpp:
(WebCore::CSSPropertyParserHelpersWorkerSafe::consumeFontFaceSrc):

Canonical link: <a href="https://commits.webkit.org/258749@main">https://commits.webkit.org/258749@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/261cc62c05dff3ec27a654fecef6e307f23c5c23

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102795 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11915 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35822 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/112052 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/172274 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/106760 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12930 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2824 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/95046 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/109728 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108569 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/9929 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/93134 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/37571 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/91779 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24656 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/79310 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/5370 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26072 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5518 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/2522 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11537 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45565 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6015 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/7261 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
r=me for last commit https://github.com/WebKit/WebKit/commit/f0ef782a516469b245ca5a534a3e7ac7e60edb54

vitorroriz committed:
font-face src list fails early if component fails
https://bugs.webkit.org/show_bug.cgi?id=250332
rdar://104039050

Reviewed by Tim Nguyen.

If one component of the list fails we should not immediately fail the parsing
of font src, since one valid component will validate the descriptor.
Therefore, we should just fail the src descriptor parsing if all components
have failed.

"A component with a parse error does not invalidate the entire descriptor if there's some other valid component present."
Reference: https://github.com/web-platform-tests/wpt/blob/e9470cde64/css/css-fonts/parsing/font-face-src-list.html

* LayoutTests/fast/css/font-face-src-parsing-expected.txt:
* LayoutTests/fast/css/font-face-src-parsing.html:

* LayoutTests/fast/css/font-face-unquoted-local-expected.html: Removed.
* LayoutTests/fast/css/font-face-unquoted-local-expected.txt: Added.
* LayoutTests/fast/css/font-face-unquoted-local.html:
This test was making use of faulty behavior of src list failing early to build the test.

* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-face-src-format-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-face-src-list-expected.txt:
* Source/WebCore/css/parser/CSSPropertyParserWorkerSafe.cpp:
(WebCore::CSSPropertyParserHelpersWorkerSafe::consumeFontFaceSrc):

Canonical link: https://commits.webkit.org/258749@main
